### PR TITLE
CUDA 11.4+ in fact supports GCC 11

### PR DESCRIPTION
**Pull request by kkm000**
GCC 10 and 11 are supported starting from CUDA v11.4.

See https://docs.nvidia.com/cuda/archive/11.4.2/cuda-installation-guide-linux/index.html

### CODE REVIEW

**Comment by kkm000** [comment]
@jtrmal, I m going to tear this apart into 2 or 3 easier PRs. It's just that a commit SHA had to be in the main repo to build in the Burrmill pipeline with fewer tweaks. And now I need to install portaudio on Travis explicitly. :)

Thanks for looking! Hope I'm on the right track!

File: src/Makefile [hunk 1]
```diff
@@ -131,69 +148,96 @@ valgrind:
...
-### Dependency list ###
-# Three base libraries that mostly everything depends on
-BMU = base matrix util
+########################################################################
+### Dependency list
+
+# 1) Binaries. A bin's main lib is listed first, extra space, then the rest of
+#    dependencies in lexicographic order.
+#    Exceptions: fgmmbin's main lib is gmm; bin does not have one.
+bin:        decoder  fstext gmm hmm lat lm transform tree
```
Comment (kkm000) [inline src/Makefile:178]: Extra space
```suggestion
bin:        decoder fstext gmm hmm lat lm transform tree
```

**Comment by kkm000** [comment]
Deat @kkm000, I am writing to you so that you remember it:

> [[build] Unlump binary-to-lib dependencies](https://github.com/kaldi-asr/kaldi/pull/4653/commits/93f519fa61ce152580f80611bec65701d439996a)
[93f519f](https://github.com/kaldi-asr/kaldi/pull/4653/commits/93f519fa61ce152580f80611bec65701d439996a)
>
>
> Extract and set true dependencies between binary and lib directories.
> The dependency of all binaries on all libraries produced bizarre
> effects.
> 
> Also, refer to tools/install_portaudio if it's a dependency but is
> missing instead of downloading, compiling and installing it during
> main Kaldi build. We do that for all other dependencies, and this
> exception made no sense.

There it is!!! I was under an impression I committed this Makefile cleanup, and was wondering why don't I see it on the master!

**Comment by jtrmal** [comment]
@kkm000  do you want me to take over this? it seems to me it's pretty much ready to go, just needs to get resoved

**Comment by kkm000** [comment]
@jtrmal, yeah, please, looks like I'm bogged down. I'd really appreciate. Sorry.

**Comment by jtrmal** [comment]
I'm closing this down, as I have adopted/moved everything to PR #4744

**Comment by kkm000** [comment]
Yay! Thanks!

**Comment by jtrmal** [comment]
no worries, thanks for the work already done!
y.

**Pull request closed without merging**